packages/wallet-frontend/src/ui/forms/Input.tsx

```diff
@@ -1,29 +1,56 @@
-import { forwardRef, useId } from 'react'
+import { type ReactNode, forwardRef, useId } from 'react'
 import type { ComponentPropsWithoutRef } from 'react'
-
 import { Label } from './Label'
 import { FieldError } from './FieldError'
+import { cx } from 'class-variance-authority'
 
-type InputProps = Omit<ComponentPropsWithoutRef<'input'>, 'className'> & {
+type InputProps = ComponentPropsWithoutRef<'input'> & {
   label?: string
   error?: string
+  addOn?: ReactNode
+  trailing?: ReactNode
+  labelHint?: ReactNode
```

Comment:
I have been considering the possibility of implementing a `trailingIcon` or `isLoading` property. My reasoning behind this is as follows:
- there may be certain inputs that require asynchronous work - the `isLoading` case
- it may be useful to include a trailing icon (such as an error badge or a show/hide button for password) - the `trailingIcon` case

My primary concern is whether or not we should allow both trailing content and a trailing icon. Should we only permit one of these trailing options (e.g., if `trailing` is present, we do not accept `trailingIcon` and vice versa)?

Reply:
It would be very unusual to have both at the same time indeed, but not impossible. I would think it is ok to allow both at the same time...